main.go

```diff
@@ -160,12 +160,20 @@ func SlightEchoHandler(writer http.ResponseWriter, request *http.Request) {
 	writer.Write(responsePayload)
 }
 
-func runWorker(c *cli.Context, n *server.Node) error {
+func runWorker(c *cli.Context, n *server.Node, g peer.ID) error { // TODO: Is there a reason all these have just c,n as arguments?
 	time.Sleep(10 * time.Second)
+	var f, release = n.Cluster.View().Lookup(c.Context) // TODO: Submit a query for ("gatewayID/lb/chan")?
+	defer release()
+	var record, err = f.Await(c.Context) // Confused because record is wrong type, so how get chan capability from record?
+	if err == nil {
+		return err
+	}
+	n.Vat.Connect(c.Context, g)
+
 	return errors.New("WORKER NOT IMPLEMENTED")
 }
 
-func waitPeers(c *cli.Context, n *server.Node) (bool, error) {
+func waitPeers(c *cli.Context, n *server.Node) (peer.ID, error) {
```

Comment:
I think this change is where you're getting thrown off-track. Remember that you're running several instances of this program, in separate processes.  Thus, each process only needs to make the decision "am _I_ the gateway, or is someone else?".

If you are the gateway, creating the capability in the worker is a simple matter of setting up the echo capnp server, and then creating a client for it (just like in the tutorial).  You don't need to export it or do anything fancy.  Ultimately you'll be sending it down the channel.

Reply:
Oh, I think I see where you're going:   you're passing the gateway's `peer.ID` to the worker, so it knows where to get the channel.  👍 

Keep in mind that there are two capabilities at play:

1. The channel (owned and exported by the gateway)
2. Each worker's `Echo` client (owned by the worker; **not** exported, but will be sent down the gateway's channel)

Does this help?

Reply:
Yeah, the part I am confused about is how the worker references the channel capability when trying to do n.Vat.Connect(). (Sorry for the spam)